src/topp/OpenSwathWorkflow.cpp

```diff
@@ -905,63 +950,148 @@ class TOPPOpenSwathWorkflow
     calibration_param.setValue("mz_extraction_window_ppm", cp_irt.ppm ? "true" : "false");
     calibration_param.setValue("im_extraction_window", cp_irt.im_extraction_window);
     calibration_param.setValue("mz_correction_function", mz_correction_function);
+    // Remove iRT params grouped with TOPPsubsection Calibration params
+    calibration_param.remove("irt_seed");
+    calibration_param.remove("irt_bins");
+    calibration_param.remove("irt_peptides_per_bin");
+    calibration_param.remove("irt_bins_nonlinear");
+    calibration_param.remove("irt_peptides_per_bin_nonlinear");
+    calibration_param.remove("tr_irt");
+    calibration_param.remove("tr_irt_nonlinear");
+    calibration_param.remove("rt_norm");
+
+    // 1) Prepare in‐memory iRT experiments for linear + nonlinear
+    OpenSwath::LightTargetedExperiment lin_irt_exp;
+    if (!irt_tr_file.empty())
+    {
+      // user‐supplied linear iRT file takes absolute precedence
+      FileTypes::Type irt_tr_type = FileHandler::getType(irt_tr_file);
+      Param irt_tsv_reader_param = TransitionTSVFile().getDefaults();
+      lin_irt_exp = loadTransitionList(irt_tr_type, irt_tr_file, irt_tsv_reader_param);
+    }
+    else if (auto_irt)
+    {
+      OPENMS_LOG_INFO << "Linear iRT Calibration: Sampling input transition experiment for " << irt_bins_lin << " bins across the RT with " << irt_pep_lin << " peptides per bin" << std::endl;
+      // sampled transtion_exp on‐the‐fly
+      lin_irt_exp = OpenSwathHelper::sampleExperiment(
+        transition_exp,
+        irt_bins_lin,
+        irt_pep_lin,
+        irt_seed,
+        true,
+        true,
+        0.4
```

Comment:
Still having difficulty understanding this parameter however not sure if it should be hardcoded?

Reply:
Maybe include some more comments in the code explaining the reasoning behind the magic constant?